staging/src/k8s.io/client-go/tools/events/interfaces.go

```diff
@@ -68,3 +69,15 @@ type EventSink interface {
 	Update(event *v1beta1.Event) (*v1beta1.Event, error)
 	Patch(oldEvent *v1beta1.Event, data []byte) (*v1beta1.Event, error)
 }
+
+// EventClient knows whether to use new or old Event API for handling events.
+type EventClient interface {
+	// StartRecordingToSink starts sending events received from the specified eventBroadcaster.
+	StartRecordingToSink(stopCh <-chan struct{})
+
+	// NewEventRecorder creates a new Event Recorder with specified name.
+	NewEventRecorder(name string) EventRecorder
```

Comment:
Let's call it `NewRecorder` for consistency with the EventBroadcaster API it will make (later migration easier).

Reply:
Done.